src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -148,16 +178,38 @@ public function addConfiguration(NodeDefinition $node)
         $builder
             ->scalarNode('secret')->isRequired()->cannotBeEmpty()->end()
             ->scalarNode('service')->end()
-            ->scalarNode('token_provider')->end()
             ->arrayNode('user_providers')
                 ->beforeNormalization()
                     ->ifString()->then(function ($v) { return [$v]; })
                 ->end()
                 ->prototype('scalar')->end()
             ->end()
             ->booleanNode('catch_exceptions')->defaultTrue()->end()
+            ->arrayNode('signature_properties')
+                ->prototype('scalar')->end()
+                ->requiresAtLeastOneElement()
+                ->info('An array of properties on your User that are used to sign the rememberme cookie. If any of these change, all existing cookies will become invalid.')
+                ->example(['email', 'password'])
+            ->end()
         ;
 
+        $tokenProviderBuilder = $builder
+            ->arrayNode('token_provider')
+                ->beforeNormalization()
+                    ->ifString()->then(function ($v) { return ['service' => $v]; })
+                ->end()
+                ->children()
+                    ->scalarNode('service')->end();
+        if (class_exists(DoctrineTokenProvider::class)) {
+            $tokenProviderBuilder
+                ->arrayNode('doctrine')
+                    ->canBeEnabled()
+                    ->children()
+                        ->scalarNode('connection')->defaultValue('default')->end()
```

Comment:
instead of hardcoding `default` as connection name (which is not guaranteed to be the name of the default DBAL connection as that name is customizable), the default value should be `null`. And when set to `null`, you should use the `database_connection` id for the reference, which is the alias pointing to the default DBAL connection.